service/project.go

```diff
@@ -662,13 +662,18 @@ func (uis *UIServer) modifyProject(w http.ResponseWriter, r *http.Request) {
 			return
 		}
 	}
-	if origProjectRef.UseRepoSettings != projectRef.UseRepoSettings {
-		if projectRef.UseRepoSettings {
-			err = projectRef.AddToRepoScope(dbUser)
-		} else {
-			err = projectRef.RemoveFromRepoScope()
+
+	// if owner/repo has changed or we're toggling repo settings off, update scope
+	if projectRef.Owner != origProjectRef.Owner || projectRef.Repo != origProjectRef.Repo ||
```

Comment:
I'm not seeing when this part would be hit. If a project ref is a branch, when would its owner/repo be different than the parent repo ref?

Reply:
origProjectRef isn't the parent repo ref, it's the project ref before the user added their updates, so this check is basically handles the case "move this branch configuration to a different owner/repo" (also this makes sure that our repo code doesn't get out of touch with legacy for now).

Reply:
ah I confused that part, but my point was more that I wouldn't expect this situation to have to change any sort of scopes. Looking in to this more, if we didn't have a naming convention for the _id of the scope we wouldn't need to change it here. Are you sure having the naming convention is worth it?